Week4/WEEK-4-SOLUTION.ipynb

```diff
@@ -0,0 +1,454 @@
+      "Fname:sunil\n",
+      "Lname:sakaliya\n",
+      "Contact:9326367372\n",
+      "Email:sunil@gmail.com\n",
+      "city:asta\n",
+      "\n",
+      "ID:4\n",
+      "Fname:ayaz\n",
+      "Lname:shaikh\n",
+      "Contact:8976620101\n",
+      "Email:ayaz@gmail.com\n",
+      "city:bardoli\n",
+      "\n",
+      "ID:5\n",
+      "Fname:sai\n",
+      "Lname:patel\n",
+      "Contact:9987120302\n",
+      "Email:sai@gmail.com\n",
+      "city:madhi\n"
+     ]
+    }
+   ],
+   "source": [
+    "cur.execute(\"select * from contact\")\n",
+    "row=cur.fetchall()\n",
+    "for i in row:\n",
+    "    print(f\"\\nID:{i[0]}\\nFname:{i[1]}\\nLname:{i[2]}\\nContact:{i[3]}\\nEmail:{i[4]}\\ncity:{i[5]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "6917d439",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[('kailash', 'shahu', 9316203288, 'NULL', '2023-08-10 07:50:20', 'insert'), ('sunil', 'sakaliya', 9034567898, 'NULL', '2023-08-10 07:50:20', 'insert'), ('dev', 'rathod', 8970564789, 'NULL', '2023-08-10 07:50:20', 'insert'), ('ayaz', 'shaikh', 8976620101, 'NULL', '2023-08-10 07:50:20', 'insert'), ('sai', 'patel', 9987120302, 'NULL', '2023-08-10 07:50:20', 'insert'), ('sunil', 'sakaliya', 8976567890, 9034567898, '2023-08-10 07:50:45', 'update'), ('dev', 'rathod', 'NULL', 8970564789, '2023-08-10 07:50:48', 'delete')]\n",
+      "\n",
+      "Fname:kailash\n",
+      "Lname:shahu\n",
+      "New-contact:9316203288\n",
+      "Old-Contact:NULL\n",
+      "Datetime:2023-08-10 07:50:20\n",
+      "Operation:insert\n",
+      "\n",
+      "Fname:sunil\n",
+      "Lname:sakaliya\n",
+      "New-contact:9034567898\n",
+      "Old-Contact:NULL\n",
+      "Datetime:2023-08-10 07:50:20\n",
+      "Operation:insert\n",
+      "\n",
+      "Fname:dev\n",
+      "Lname:rathod\n",
+      "New-contact:8970564789\n",
+      "Old-Contact:NULL\n",
+      "Datetime:2023-08-10 07:50:20\n",
+      "Operation:insert\n",
+      "\n",
+      "Fname:ayaz\n",
+      "Lname:shaikh\n",
+      "New-contact:8976620101\n",
+      "Old-Contact:NULL\n",
+      "Datetime:2023-08-10 07:50:20\n",
+      "Operation:insert\n",
+      "\n",
+      "Fname:sai\n",
+      "Lname:patel\n",
+      "New-contact:9987120302\n",
+      "Old-Contact:NULL\n",
+      "Datetime:2023-08-10 07:50:20\n",
+      "Operation:insert\n",
+      "\n",
+      "Fname:sunil\n",
+      "Lname:sakaliya\n",
+      "New-contact:8976567890\n",
+      "Old-Contact:9034567898\n",
+      "Datetime:2023-08-10 07:50:45\n",
+      "Operation:update\n",
+      "\n",
+      "Fname:dev\n",
+      "Lname:rathod\n",
+      "New-contact:NULL\n",
+      "Old-Contact:8970564789\n",
+      "Datetime:2023-08-10 07:50:48\n",
+      "Operation:delete\n"
+     ]
+    }
+   ],
+   "source": [
+    "cur.execute(\"select * from details_log\")\n",
+    "row1=cur.fetchall()\n",
+    "print(row1)\n",
+    "for i in row1:\n",
```

Comment:
why we use for loop ?

Reply:
for loops are used when you have a block of code which you want to repeat a fixed number of times.